src/indexer/merger.rs

```diff
@@ -98,27 +96,23 @@ pub struct IndexMerger {
 
 fn compute_min_max_val(
     u64_reader: &impl FastFieldReader<u64>,
-    max_doc: DocId,
-    delete_bitset_opt: Option<&DeleteBitSet>,
+    segment_reader: &SegmentReader,
 ) -> Option<(u64, u64)> {
-    if max_doc == 0 {
+    if segment_reader.max_doc() == 0 {
         None
```

Comment:
Please use the return early pattern on that kind of stuff.
```
 if segment_reader.max_doc() == 0 {
        None
}

if segment_reader.alive_bitset().is_none() {
    // no deleted documents,
    // we can use the previous min_val, max_val.
    return Some((u64_reader.min_value(), u64_reader.max_value()));
}

// some deleted documents,
            // we need to recompute the max / min
            minmax(
                segment_reader
                    .doc_ids_alive()
                    .map(|doc_id| u64_reader.get(doc_id)),
```

Reply:
done, I also prefer it that way (easier to read due to less indentation)